### PR TITLE
Add cargo-deny to CI

### LINKED ISSUE #883: Add cargo-deny to CI
https://github.com/EmbarkStudios/cargo-deny

An example from another project: https://github.com/bytecodealliance/wasmtime/pull/2519

### PR DESCRIPTION
**Pull request by brightly-salty**
Fixes #883 

Adds cargo-deny to CI, to run on every pull request
Also adds license information to Cargo.toml, and renames license file (presumed typo)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name = "gleam"
 version = "0.13.2"
 authors = ["Louis Pilfold <louis@lpil.uk>"]
 edition = "2018"
+license-file = "LICENSE"
```
Comment (lpil) [inline Cargo.toml:6]: Can we use LICENCE rather than LICENSE? I'm not American :)

**Review by lpil — COMMENTED** [review]
Fab, thank you. I've left one comment online

File: Cargo.toml (hunk 1 above)
Comment (brightly-salty) [reply Cargo.toml:6]: Sure! (I am, so I presumed it to be a typo. I actually just used the spelling of cargo-deny, was always unsure before now!).

**Review by lpil — APPROVED** [review]
Thank you

**Pull request merged**